### PR TITLE
FEATURE: Send a message to new TL2 users and point them to the advanced tutorial

### PR DESCRIPTION
**Pull request by romanrizzi**
Screen Shot 2019-11-11 at 18 09 57

### CODE REVIEW

File: plugins/discourse-narrative-bot/autoload/jobs/send_advanced_tutorial_message.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Jobs
+  class SendAdvancedTutorialMessage < ::Jobs::Base
+    def execute(args)
+      user = User.find_by(id: args[:user_id])
+      raise Discourse::InvalidParameters.new(:user_id) if user.nil?
+
+      discobot_user = User.find(-2)
```
Comment (ZogStriP) [inline plugins/discourse-narrative-bot/autoload/jobs/send_advanced_tutorial_message.rb]: You can instead use

```suggestion
      discobot_user = DiscourseNarrativeBot::Actions.discobot_user
```

File: plugins/discourse-narrative-bot/spec/jobs/send_advanced_tutorial_message_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+RSpec.describe Jobs::SendAdvancedTutorialMessage do
+  before do
+    Jobs.run_immediately!
+    SiteSetting.discourse_narrative_bot_enabled = true
+  end
+
+  it 'sends a message to the promoted user' do
+    user = Fabricate(:user)
+    discobot_user = User.find(-2)
```
Comment (eviltrout) [inline plugins/discourse-narrative-bot/spec/jobs/send_advanced_tutorial_message_spec.rb]: I'd also update this one to use the constant @ZogStriP  suggested.

**Comment by romanrizzi** [comment]
@eviltrout @ZogStriP I replaced all occurrences of `User.find(-2)` with that method call. 👍

**Comment by coding-horror** [comment]
wait why is this coming from discobot? What if discobot is disabled? I don't think the TL1 "congrats" PM comes from discobot?

**Comment by romanrizzi** [comment]
> I don't think the TL1 "congrats" PM comes from discobot?

Good catch, @coding-horror! I changed it to system user instead of discobot.

> What if discobot is disabled?

If this happens, the message won't be sent since there won't be an advanced tutorial either.

File: plugins/discourse-narrative-bot/spec/jobs/send_advanced_tutorial_message_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+RSpec.describe Jobs::SendAdvancedTutorialMessage do
+  before do
+    Jobs.run_immediately!
+    SiteSetting.discourse_narrative_bot_enabled = true
+  end
+
+  it 'sends a message to the promoted user' do
+    user = Fabricate(:user)
+    discobot_user = Discourse.system_user
```
Comment (ZogStriP) [inline plugins/discourse-narrative-bot/spec/jobs/send_advanced_tutorial_message_spec.rb]: Technically, `system_user` is not `discobot` ;)

**Pull request merged**